File: tests/fixtures/patterns/php/positive.php
<?php
// Positive fixture: each snippet should trigger the named pattern.

// php.code_exec.eval
function trigger_eval($code) {
    eval($code);
}

// php.code_exec.create_function
function trigger_create_function() {
    $fn = create_function('$a', 'return $a * 2;');
}

// php.code_exec.preg_replace_e
function trigger_preg_replace_e($input) {
    preg_replace('/test/e', 'strtoupper("$1")', $input);
}

// php.code_exec.assert_string
function trigger_assert($code) {
    assert("strlen('$code') > 0");
}

// php.cmdi.system
function trigger_system($cmd) {
    system($cmd);
}

// php.deser.unserialize
function trigger_unserialize($data) {
    unserialize($data);
}

// php.sqli.query_concat
function trigger_sql_concat($user) {
    mysql_query("SELECT * FROM users WHERE name = '" . $user . "'");
}

// php.path.include_variable
function trigger_include($path) {
    include $path;
}

// php.crypto.md5
function trigger_md5($data) {
    md5($data);
}

// php.crypto.sha1
function trigger_sha1($data) {
    sha1($data);
}

// php.crypto.rand
function trigger_rand() {
    $r = rand();
}

File: tests/fixtures/patterns/python/positive.py
# Positive fixture: each snippet should trigger the named pattern.

import os
import subprocess
import pickle
import yaml
import hashlib

# py.code_exec.eval
def trigger_eval(data):
    result = eval(data)

# py.code_exec.exec
def trigger_exec(code):
    exec(code)

# py.code_exec.compile
def trigger_compile(code):
    co = compile(code, "<string>", "exec")

# py.cmdi.os_system
def trigger_os_system(cmd):
    os.system(cmd)

# py.cmdi.os_popen
def trigger_os_popen(cmd):
    os.popen(cmd)

# py.cmdi.subprocess_shell
def trigger_subprocess_shell(cmd):
    subprocess.run(cmd, shell=True)

# py.deser.pickle_loads
def trigger_pickle(data):
    obj = pickle.loads(data)

# py.deser.yaml_load
def trigger_yaml(data):
    obj = yaml.load(data)

# py.sqli.execute_format
def trigger_sql_concat(cursor, user):
    cursor.execute("SELECT * FROM users WHERE name = '" + user + "'")

# py.crypto.md5
def trigger_md5(data):
    hashlib.md5(data)

# py.crypto.sha1
def trigger_sha1(data):
    hashlib.sha1(data)
